lib/unleash/constraint.rb

```diff
@@ -1,26 +1,62 @@
+require 'date'
+require 'unleash/constraints/contains_constraints'
+require 'unleash/constraints/date_constraints'
+require 'unleash/constraints/numeric_constraints'
+require 'unleash/constraints/semver_constraints'
+require 'unleash/constraints/string_constraints'
+
 module Unleash
   class Constraint
-    attr_accessor :context_name, :operator, :values
+    attr_accessor :context_name, :operator, :value, :inverted, :case_insensitive
 
-    VALID_OPERATORS = ['IN', 'NOT_IN'].freeze
+    VALID_OPERATORS = [
+      ConstraintMatcher::ContainsConstraint::OPERATORS,
+      ConstraintMatcher::StringConstraint::OPERATORS,
+      ConstraintMatcher::NumericConstraint::OPERATORS,
+      ConstraintMatcher::DateConstraint::OPERATORS,
+      ConstraintMatcher::SemverConstraint::OPERATORS
+    ].flatten.freeze
 
-    def initialize(context_name, operator, values = [])
+    def initialize(context_name, operator, value = [], inverted: false, case_insensitive: false)
```

Comment:
Since the constructor changed, it's initialization with all parameters should also be tested in the `feature_toggle_spec.rb`

Reply:
Also, methods/constructors with more than 3 parameters are usually frowned upon. It's late so I wont' come with a suggestion right now 🙈 .

Reply:
Yes! I'll extend the tests here! I'm open to suggestions on the method parameters but a value object felt like overkill here. Hash maybe?

Reply:
I've been thinking about this. I don't think a Hash is better here, it makes the values opaque to the caller and requires extra validation layers and two of the parameters are optionals, for most cases the calling code shouldn't be excessively long. I'm tempted to just leave this as is but if anyone has a better idea than Hashes (or can tell me I'm wrong about their usage here!) I'm open

Reply:
I feel quite uneasy about the hybrid approach here. (using hashes for the new values, but not for everything. But again, maybe we can just leave as is for the params w/o default values?

Reply:
I have thought about this one way too much - yeah the hash suggestion isn't right here. I want to leave the new default values in for the inverted and case Insensitive, mostly because they won't be needed 95% of the time (caseInsensitive won't do anything outside of the string operators and most constraints won't need inversion because they can be expressed as their equal and opposite constraint - !GTE == LT)